src/main/fc/fc_msp.c

```diff
@@ -1866,7 +1867,11 @@ static bool processInCommand(uint8_t cmdMSP)
         masterConfig.gyro_sync_denom = read8();
         masterConfig.pid_process_denom = read8();
         masterConfig.motorConfig.useUnsyncedPwm = read8();
-        masterConfig.motorConfig.motorPwmProtocol = read8();
+#ifdef USE_DSHOT
+        masterConfig.motorConfig.motorPwmProtocol = constrain(read8(), 0, PWM_TYPE_DSHOT150);
+#else
+        masterConfig.motorConfig.motorPwmProtocol = constrain(read8(), 0, PWM_TYPE_BRUSHED);
+#endif
```

Comment:
Isn't it better to use separate enum (PWM_TYPE_COUNT ?) to get lastvalue+1 automagically ?